src/EditorFeatures/Core/EncapsulateField/AbstractEncapsulateFieldCommandHandler.cs

```diff
@@ -115,20 +115,25 @@ public bool ExecuteCommand(EncapsulateFieldCommandArgs args, CommandExecutionCon
         {
             var finalSolution = await result.GetSolutionAsync(cancellationToken).ConfigureAwait(false);
 
+            // Move to the UI thread to apply changes
+            await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
```

Comment:
So given that PreviewChangesAsync is now well behaved, then just switch right before we take the undo transaction? Then your comment for the ConfigureAwait(true) isn't as critical, and nobody else can screw up in the middle.